### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by helloandrewpark**

### CODE REVIEW

File: src/identity-handler.js
Comment (alexs-mparticle) [inline src/identity-handler.js:28]: I'm not sure where you're setting `userIdentificationType` or `settings` for that matter. You should probably have a safety check before this line to make sure `settings` and `userIdentificationType` are not null.

Comment (helloandrewpark) [reply src/identity-handler.js:28]: Is it possible for either of those to be null while `mParticleUser.getUserIdentities().userIdentities` returns valid identities? if so is it better to get the identity type from `mParticleUser.getUserIdentities().userIdentities` somehow?

Comment (rmi22186) [reply src/identity-handler.js:28]: I realize there is a bug in the file and put up a PR with a few [bugfixes](https://github.com/mparticle-integrations/mparticle-javascript-integration-example/pull/21) in our integration-example repo, but specifically it's this [commit](https://github.com/mparticle-integrations/mparticle-javascript-integration-example/pull/21/commits/b2c5294b430941b14047cafa4908e48f022d292c).  

once this is merged, you can rebase the updates.

you can then follow branch's lead on this line [here](https://github.com/BranchMetrics/mparticle-javascript-integration-branch/blob/master/src/initialization.js#L4).  you set the forwarder settings onto the `common` module, which is then available here via `this.common.settings` as they've done.

i'll ping this thread once the PR above is merged

Comment (rmi22186) [reply src/identity-handler.js:28]: another thing - you said you took branch's lead on this. they specifically called out a `userIdentificaitonType` setting that is available in Branch.  Does userleap have something like this?  or is there just a userId.  i assume that would match to `customerId` but since i'm not familiar with your platform, some guidance would be helpful.

it looks like the settings discussed were only `apiKey` and `environmentId`.  please confirm.

Comment (rmi22186) [reply src/identity-handler.js:28]: btw the above PR was merged so you could rebase it in

Comment (helloandrewpark) [reply src/identity-handler.js:28]: Simplifying the implementation, let me know if this is valid

File: test/end-to-end-testapp/build/compilation.js
Comment (rmi22186) [inline test/end-to-end-testapp/build/compilation.js:1]: This build file should not be included

Comment (helloandrewpark) [reply test/end-to-end-testapp/build/compilation.js]: Thanks @rmi22186, just removed!

File: package.json
Comment (rmi22186) [inline package.json:3]: technically this is V1.0. of your integration.  it was just 1.0.2 of the example at the time you forked it

File: dist/UserLeap-Kit.iife.js
Comment (rmi22186) [inline dist/UserLeap-Kit.iife.js:1]: totally your call, but we find it best to commit all src code, then in once approved, do another commit to build the files. while this doesn't guarantee that you will build the file before committing it as part of step 2, it reminds you to.  this applies to both `UserLeap-Kit.iife.js` and `UserLeap-Kit.common.js` above

File: src/initialization.js
Comment (rmi22186) [inline src/initialization.js:29]: we queue events in `eventQueue` if UserLeap is not yet available. in order to capture these events, you should include lines 24-33 in this script.onload function.

as I'm looking at the example code on the left, I don't see any UserLeap.initalize method here.  is that not necessary because you are embedding the envId in your script src on line 25?

File: test/end-to-end-testapp/settings.js
Comment (rmi22186) [inline test/end-to-end-testapp/settings.js:2]: if you are okay with including this code in your repo, then it's fine to keep this, but if this is a sensitive key,  you should use an environment variable to keep things like this a secret.  but i know API keys are public so this may not matter for you

**Review by rmi22186 — CHANGES_REQUESTED** [review]
looking good!  i did another review of the full PR to jog my memory and caught a few other things.

File: src/initialization.js
Comment (helloandrewpark) [reply src/initialization.js:29]: I'm sorry, I don't fully understand how the `eventQueue` related to 24-33. 24-29 is related to setting the <script> properties that loads our js file.

And for initialization, that is handled by our IIFE in our js file. We set the needed properties (envId in 22) in window.UserLeap and everything references that

Comment (rmi22186) [reply src/initialization.js:29]: It's very possible that UserLeap has its own concept of an event queue.  We implement one in case our partners don't have one.  Basically, our kit `eventQueue` is abstracted away, but let's say mParticle loads and the user starts firing off events before UserLeap loads.  Unless UserLeap has its own event queue, or you iterate over our provided eventQueue in the kit, these events will get lost.

If UserLeap has its own internal way of handling the event queue then the above comment is moot.

Comment (rmi22186) [reply src/initialization.js:29]: Lines 25-31 in the previous commented code show you how to implement our kit `eventQueue`

Comment (helloandrewpark) [reply src/initialization.js:29]: ah got it, yes we have our own event queue that records calls to our SDK before the JS file has been downloaded

File: package.json
Comment (rmi22186) [inline package.json:2]: Do you currently have any other npm packages that are scoped to your company name?  if so, you should scope this to your userleap namespace, ie @userleap/mparticle-web-kit or similar. this way for people who install it via npm it makes more sense rather than this generic name here.  
once it's up on npm, we include it here in our docs page. you'll note that most of these are @mparticle/ but since you can't scope it to us since you maintain it, you'll have to use your namespace or other.  branch uses `branch-mparticle-web-kit` as an example.  it's up to you how you want to name the npm package.

**Review by rmi22186 — COMMENTED** [review]
once the last comment is addressed regarding your package name, this looks great!

**Review by rmi22186 — APPROVED** [review]

**Review by alexs-mparticle — APPROVED** [review]

**Review by gongstr — APPROVED** [review]

**Pull request merged**
